windows_docker_resources/Dockerfile.msvc2019

```diff
@@ -0,0 +1,151 @@
+
+# Files to download
+ADD `
+    # Visual Studio Build Tools and .Net SDK`
+    https://aka.ms/vs/16/release/vs_BuildTools.exe`
+    `
+    # For extracting .7z files `
+    https://www.7-zip.org/a/7z1900-x64.exe `
+    `
+    # Qt5 online installer`
+    http://download.qt.io/official_releases/online_installers/qt-unified-windows-x86-online.exe `
+    `
+    # Download Destination `
+    C:\TEMP\
+
+# Downloading these separately because hyperlinks are versioned. Renaming to fixed names
+ADD https://github.com/ADLINK-IST/opensplice/releases/download/OSPL_V6_9_190925OSS_RELEASE/PXXX-VortexOpenSplice-6.9.190925OSS-HDE-x86_64.win-vs2019-installer.zip C:\TEMP\OpenSplice.zip
+
+# OpenSSL
+ADD https://slproweb.com/download/Win64OpenSSL-1_0_2u.exe C:\TEMP\Win64OpenSSL.exe
+
+# OpenCV
+ADD https://github.com/ros2/ros2/releases/download/opencv-archives/opencv-3.4.6-vc16.VS2019.zip C:\TEMP\opencv.zip
+
+# Python 3.7
+ADD https://www.python.org/ftp/python/3.7.6/python-3.7.6-amd64.exe C:\TEMP\python-37.exe
+
+# Custom choco packages
+ADD https://github.com/ros2/choco-packages/releases/download/2019-10-24/asio.1.12.1.nupkg C:\TEMP\asio.nupkg
+ADD https://github.com/ros2/choco-packages/releases/download/2019-10-24/cunit.2.1.3.nupkg C:\TEMP\cunit.nupkg
+ADD https://github.com/ros2/choco-packages/releases/download/2019-10-24/eigen.3.3.4.nupkg C:\TEMP\eigen.nupkg
+ADD https://github.com/ros2/choco-packages/releases/download/2019-10-24/log4cxx.0.10.0.nupkg C:\TEMP\log4cxx.nupkg
+ADD https://github.com/ros2/choco-packages/releases/download/2019-10-24/tinyxml-usestl.2.6.2.nupkg C:\TEMP\tinyxml.nupkg
+ADD https://github.com/ros2/choco-packages/releases/download/2019-10-24/tinyxml2.6.0.0.nupkg C:\TEMP\tinyxml2.nupkg
+
+# xmllint files
+ADD https://www.zlatkovic.com/pub/libxml/64bit/libxml2-2.9.3-win32-x86_64.7z C:\TEMP\libxml2.7z
+ADD https://www.zlatkovic.com/pub/libxml/64bit/zlib-1.2.8-win32-x86_64.7z C:\TEMP\zlib.7z
+ADD https://www.zlatkovic.com/pub/libxml/64bit/iconv-1.14-win32-x86_64.7z C:\TEMP\iconv.7z
+
+# Restore the default Windows shell for correct batch processing.
+SHELL ["cmd", "/S", "/C"]
+
+# 3010 is an acceptable exit code, but it will confuse docker
+RUN C:\TEMP\vs_BuildTools.exe --quiet --wait --norestart --includeRecommended `
+  --add Microsoft.Net.Component.4.8.SDK `
+  --add Microsoft.VisualStudio.Workload.VCTools `
+  --add Microsoft.Component.MSBuild `
+  --add Microsoft.VisualStudio.Component.VC.CLI.Support `
+  || IF "%ERRORLEVEL%"=="3010" EXIT 0
+
+RUN C:\TEMP\python-37.exe /quiet `
+  TargetDir=C:\Python37 `
+  PrependPath=1 `
+  Include_debug=1 `
+  Include_symbols=1
+
+# Install Chocolatey by powershell script
+RUN powershell -noexit "Set-ExecutionPolicy Bypass -Scope Process -Force; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))"
+
+# choco installs
+RUN choco install -y cmake curl git KB3035131 KB3033929 KB2919442 vcredist2013 vcredist140 cppcheck
```

Comment:
Can we document what these KB payloads are and why they are needed?

Reply:
choco installs are sometimes flakey, and these KB's were needed at one point. They don't look like they are needed anymore with 1909. Removed